packages/components/src/components/dropdown/dropdown.tsx

```diff
@@ -1,18 +1,72 @@
-import classNames from 'classnames';
-import PropTypes from 'prop-types';
 import React from 'react';
 import ReactDOM from 'react-dom';
+import classNames from 'classnames';
 import { CSSTransition } from 'react-transition-group';
 import { mobileOSDetect, getPosition } from '@deriv/shared';
-import { listPropType, findNextFocusableNode, findPreviousFocusableNode } from './dropdown';
-import Items from './items.jsx';
-import DisplayText from './display-text.jsx';
+import { TList, findNextFocusableNode, findPreviousFocusableNode, TListItem } from './utility';
+import Items from './items';
+import DisplayText from './display-text';
 import Text from '../text/text';
-import { useBlockScroll, useOnClickOutside } from '../../hooks';
+import { IClickEvent, useBlockScroll, useOnClickOutside } from '../../hooks';
 import ThemedScrollbars from '../themed-scrollbars/themed-scrollbars';
 import Icon from '../icon/icon';
 
-const DropdownList = React.forwardRef((props, list_ref) => {
+type TDropdown = {
+    className?: string;
+    classNameDisplay?: string;
+    classNameHint?: string;
+    classNameItems?: string;
+    classNameLabel?: string;
+    disabled?: boolean;
+    error?: string;
+    handleBlur?: {
+        (e: React.FocusEvent<any>): void;
+        <T = string | any>(fieldOrEvent: T): T extends string ? (e: any) => void : void;
+    };
+    has_symbol?: boolean;
+    hint?: string;
+    initial_offset?: number;
+    is_align_text_left?: boolean;
+    is_alignment_left?: boolean;
+    is_alignment_top?: boolean;
+    is_large?: boolean;
+    is_nativepicker_visible?: boolean;
+    is_nativepicker?: boolean;
+    label?: string;
+    list_height?: string;
+    list_portal_id?: string;
+    list: TList;
+    name?: string;
+    no_border?: boolean;
+    onChange?: (e: { target: { name: string; value: string } }) => void;
+    onClick?: () => void;
+    placeholder?: string;
+    suffix_icon?: string;
+    test_id?: string;
+    value?: string | number;
+};
+
+type TDropdownList = {
+    classNameItems?: string;
+    classNameLabel?: string;
+    handleSelect: (item: TListItem) => void;
+    has_symbol?: boolean;
+    initial_offset?: number;
+    is_align_text_left?: boolean;
+    is_alignment_left?: boolean;
+    is_alignment_top?: boolean;
+    is_large?: boolean;
+    is_list_visible: boolean;
+    list: TList;
+    nodes: any;
```

Comment:
@niloofar-deriv could you please try React.ReactNode[] or NodeList here instead of any?🙂

Reply:
@maryia-deriv thanks for the suggestions but none of them works disappointed